### DISCUSSION TITLE
Don't print sensitive fixtures on failure

### DISCUSSION THREAD

**Discussion by ashwin153** [pytest-dev/pytest#8910] [Q&A] [answered]
I'm using the [pytest-securestore](https://github.com/gregfitch/pytest-securestore) to store my integration test secrets (API keys, passwords, etc.). Secrets are stored in an AES-encrypted, version-controlled file that is decrypted by the plugin at runtime and injected into the tests that need them. This works great; however, whenever a test fails the contents of this file are printed to the console. For example, here is the (redacted) output that pytest produces on failure.

```
=============================================== FAILURES ===============================================
______________________________________________ test_email ______________________________________________

store = {'action_discord_webhook': '...', 'output_email_password': '...', 'output_email_sender': '...'}

    def test_email(store):
        """Verify that data can be attached to emails."""
>       email.compute(Context(overrides={
            "output_email_password": store.get("output_email_password"),
            "output_email_sender": store.get("output_email_sender"),
        }))
```

Is there an argument that I can pass to pytest to disable printing fixtures on failure?

**Comment by nicoddemus** [comment, marked as answer]
Currently there isn't, see https://github.com/pytest-dev/pytest/issues/8613 for a discussion on this topic and suggestions on how to workaround this.

**Comment by mganisin** [comment]
Just in case someone finds this (as I did), this plugin can help: https://github.com/mganisin/pytest-mask-secrets
